gms/gossiper.cc

```diff
@@ -2134,12 +2134,16 @@ void gossiper::build_seeds_list() {
     }
 }
 
-future<> gossiper::add_saved_endpoint(inet_address ep, permit_id pid) {
+future<> gossiper::add_saved_endpoint(inet_address ep, loaded_endpoint_state loaded_state, permit_id pid) {
     if (ep == get_broadcast_address()) {
         logger.debug("Attempt to add self as saved endpoint");
         co_return;
     }
 
+    if (!loaded_state.host_id) {
+        on_internal_error(logger, format("can't find host_id for ep {} for adding saved endpoint", ep));
+    }
+
```

Comment:
Could talking to an older node cause this to trigger?

Reply:
Not because it's old, since we had set the host_id in system.peers forever.
Only an exception in this process could have left the state in system.peers with no host_id (see https://github.com/scylladb/scylladb/pull/16376).
If we somehow hit this internal error, we should fix system.peers manually.

Reply:
ok